vortex-array/src/array/primitive/compute/patch.rs

```diff
@@ -36,3 +36,38 @@ fn patch_with_sparse(array: &PrimitiveArray, patch: &SparseArray) -> VortexResul
         ).into_array())
     })
 }
+
+pub fn patch_owned(array: PrimitiveArray, patch: &dyn Array) -> VortexResult<PrimitiveArray> {
```

Comment:
Replace this with. Or move this into PrimitiveArray itself so you can access buffer/validity without into_owned_children
```
impl PrimitiveArray {
    pub fn patch(self, ...) { ... }
}
```